Add --markdown output option

twang2218 opened the pull request:
## Description

Sometimes, we need the raw markdown output, such as another library consume the `tldr`'s output. For example, the [tldr plugin for Zazu](https://github.com/twang2218/zazu-tldr) need to have parsable output for processing.

So be able to have a raw markdown output would be handy.

I add `--markdown` or `-m` for short, for the raw markdown output option.

## Checklist

Please review this checklist before submitting a pull request.

- [x] Code compiles correctly
- [ ] Created tests, if possible
- [x] All tests passing (`npm run test:all`)
- [x] Extended the README / documentation, if necessary

Signed-off-by: Tao Wang <twang2218@gmail.com>

waldyrious commented:
Thanks for the contribution! @agnivade will likely have the final word on this, but personally I'd prefer if such an option was called `--raw`.

twang2218 commented:
Originally it is `--raw`, but I'm not sure the `raw` is clear enough to indicate that it's actually a markdown format out, so I replace it with `--markdown`. But both works for me.

waldyrious commented:
Sure. Let's wait and see what @agnivade thinks.

agnivade commented on bin/tldr:
```diff
@@ -92,7 +93,11 @@ if (program.list) {
   tldr.render(program.render);
 } else if (program.args.length >= 1) {
-  tldr.get(program.args, program);
+  if (program.markdown) {
```
No need to specify an if condition here. We are already passing the `program` variable to `tldr.get` which contains the markdown option. Let us do the final calculation in the `renderContent` function.

agnivade commented:
Please leave out the other changes. Make the changes in `tldr.js` that I have mentioned. Of course adding the `--markdown` option in tldr file and the autocompletion file will still be there. Rest can be left out.

twang2218 commented:
@agnivade Yes, it is much cleaner in your implementation, I changed the code by using `options` arguments, and removed the accidentally committed `package.json` file.

Could you have a look, is there anything else should I change?

agnivade commented on lib/tldr.js:
```diff
@@ -24,8 +24,8 @@ exports.listAll = (singleColumn) => {
-exports.get = (commands) => {
-  printBestPage(commands.join('-'));
+exports.get = (commands, options) => {
+  printBestPage(commands.join('-'), options);
```
Please add this line inside printBestPage function to convert to empty dictionary if `options` is not passed.

```
options = options || {};
```

to fix the travis error.

twang2218 replied:
Ok, it's done. And Travis CI is happy now. 😺
